### ISSUE TITLE
Fetch info for all streams in advance

### ISSUE THREAD

**Issue by theastropath** [theastropath/StreamDetective#14]
We should be searching for all the streams in one big request before we start trying to look for new streams or anything.

Once we have all the stream info, we can post-process it, then do the lookups as we do today.

The Twitch API can take up to 100 games or 100 usernames in a given lookup, so they can all be combined into one lookup (or probably two - one to look up all the streams by game, and another to look up all the individual streamers)

**Comment by Die4Ever** [comment]
we find streams with No Tags sometimes, I was thinking when we're about to send notifications, we could check all the streams in the newStreams array and re-fetch any that have no tags to see if they got any yet, and if we do this fetch all then there might've been enough time passed in between that we might actually get something?

**Comment by theastropath** [comment]
Fixed in acde2d459371e9338e61cc3436cecd0a3c0f7a0a

Other issue with No Tags pulled out to new issue
